### PR TITLE
OKTA-744724 | Developer Doc: Scroll Position Issue

### PR DESCRIPTION
**Pull request by madhurishivakumar-okta**
## Description:
- **What's changed?**  
  - Introduces changes to handle the scroll position when `<RouterLink />` is clicked, ensuring the user is taken to the correct section of the page.
  - Includes a fix for the accurate calculation of the active section's visible area.
- **Is this PR related to a Monolith release?**  No

### Resolves:

* [OKTA-744724](https://oktainc.atlassian.net/browse/OKTA-744724)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/layouts/Layout.vue [hunk 1]
```diff
@@ -213,11 +213,6 @@ export default {
     this.$on("toggle-tree-nav", event => {
       that.appContext.isTreeNavMobileOpen = event.treeNavOpen;
     });
-    
-    // Delay height adjustment on mount to ensure the header element is fully rendered.
-    setTimeout(() => {
```
Comment (madhurishivakumar-okta) [inline packages/@okta/vuepress-theme-prose/layouts/Layout.vue:218]: Moved this section to `HeaderBanner` to ensure the headerHeight is re-calculated only when the banner is present.

File: packages/@okta/vuepress-theme-prose/enhanceApp.js [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
-  options,
-  router,
-  siteData
-}) => {
+function scrollToAnchor(to) {
+  const targetAnchor = to.hash.slice(1);
+  const targetElement = document.getElementById(targetAnchor);
+
+  if (targetElement) {
+    const scrollToPosition = targetElement.offsetTop - 32;
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/enhanceApp.js:10]: What is 32 and will it work when banner is hidden?
I guess that's `LAYOUT_CONSTANTS.ANCHOR_TOP_MARGIN`

Comment (madhurishivakumar-okta) [reply packages/@okta/vuepress-theme-prose/enhanceApp.js:10]: Yes, It will work when the banner is hidden. The 32 is `LAYOUT_CONSTANTS.ANCHOR_TOP_MARGIN`.

Comment (emikhasyak-okta) [reply packages/@okta/vuepress-theme-prose/enhanceApp.js:10]: Can you please update it to `LAYOUT_CONSTANTS.ANCHOR_TOP_MARGIN`?

Comment (madhurishivakumar-okta) [reply packages/@okta/vuepress-theme-prose/enhanceApp.js:10]: I'll not be able to access the constants from .vue file here so, I left it as 32. I can create a separate const file for this.

**Review by emikhasyak-okta — APPROVED** [review]
LGTM! 🚢

**Pull request merged**